superset/assets/src/explore/components/controls/AdhocFilterControl.jsx

```diff
@@ -223,7 +223,7 @@ export default class AdhocFilterControl extends React.Component {
       } else if (option instanceof AdhocMetric) {
         return { ...option, filterOptionName: '_adhocmetric_' + option.label };
       }
-      return null;
+      return '';
```

Comment:
Do we really want to return an empty string here?

Reply:
The caller for these 2 `optionsForSelect` functions expecting a string array result, but don't handle `null`. if caller brought result to do sort other exception will throw.

i come up another solution that combines filter (empty metrics) + map.